tools/c7n_terraform/c7n_terraform/provider.py

```diff
@@ -0,0 +1,92 @@
+# Copyright The Cloud Custodian Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+from unittest.mock import MagicMock
+
+from c7n.registry import PluginRegistry
+from c7n.provider import Provider, clouds
+
+from c7n_terraform.resources.resource_map import ResourceMap
+
+
+class Session:
+    """Base class for API repository for a specified Cloud API."""
+
+    def __init__(self, **kwargs):
+        pass
+
+    def __repr__(self):
+        """The object representation.
+
+        Returns:
+            str: The object representation.
+        """
+        return "<tf-session>"
+
+    def client(self, service_name, version, component, **kw):
+
+        return MagicMock()
```

Comment:
this seems odd, the client invocation style here is specific to gcp provider

Reply:
A lot of this branch was heavily influenced by the c7n_gcp module, which I used as my starting point. Knowing more now than I did a month ago when I started I'll do a second pass to clean up a lot of these things

Reply:
My biggest challenge is a lot of custodian assumes there's some form of client / session which implications of what methods are available in that session. Peeling that back has taken more effort than I expected

Reply:
there shouldn't be any client construction outside of the code in this provider, the rest of the cross provider machinery should just treat the session as an opaque blob.